ln-resource-mgr/src/forward_manager.rs

```diff
@@ -159,44 +159,146 @@ impl ForwardManagerImpl {
+        let outgoing_reputation = outgoing_channel
+            .outgoing_direction
+            .outgoing_reputation(forward.added_at)?;
+
+        // Bucket resources available on the outgoing channel.
+        let outgoing_resources = ResourceValues {
+            general_bucket: BucketResources {
+                slots_used: self.htlcs.bucket_in_flight_count(
                     ChannelFilter::OutgoingChannel(forward.outgoing_channel_id),
+                    ResourceBucketType::General,
                 ),
-                // The underlying simulation is block height agnostic, and starts its routes with a height of zero, so
-                // we can just use the incoming expiry to reflect "maximum time htlc can be held on channel", because
-                // we're calculating expiry_in_height - 0.
-                htlc_risk: self
-                    .htlcs
-                    .htlc_risk(forward.fee_msat(), forward.expiry_in_height),
+                slots_available: outgoing_channel
+                    .outgoing_direction
+                    .general_bucket
+                    .slot_count,
+                liquidity_used_msat: self.htlcs.bucket_in_flight_msat(
+                    ChannelFilter::OutgoingChannel(forward.outgoing_channel_id),
+                    ResourceBucketType::General,
+                ),
+                liquidity_available_msat: outgoing_channel
+                    .outgoing_direction
+                    .general_bucket
+                    .liquidity_msat,
+            },
+            congestion_bucket: BucketResources {
+                slots_used: self.htlcs.bucket_in_flight_count(
+                    ChannelFilter::OutgoingChannel(forward.outgoing_channel_id),
+                    ResourceBucketType::Congestion,
+                ),
+                slots_available: outgoing_channel
+                    .outgoing_direction
+                    .congestion_bucket
+                    .slot_count,
+                liquidity_used_msat: self.htlcs.bucket_in_flight_msat(
+                    ChannelFilter::OutgoingChannel(forward.outgoing_channel_id),
+                    ResourceBucketType::Congestion,
+                ),
+                liquidity_available_msat: outgoing_channel
+                    .outgoing_direction
+                    .congestion_bucket
+                    .liquidity_msat,
+            },
+        };
+
+        Ok(AllocationCheck {
+            reputation_check: ReputationCheck {
+                incoming_reputation: ReputationValues {
+                    reputation: incoming_reputation,
+                    revenue_treshold: outgoing_treshold,
+                    in_flight_total_risk: self.htlcs.channel_in_flight_risk(
+                        ChannelFilter::IncomingChannel(forward.incoming_ref.channel_id),
+                    ),
+                    htlc_risk: self
+                        .htlcs
+                        .htlc_risk(forward.fee_msat(), forward.expiry_in_height),
```

Comment:
For incoming reputation we want to look at the outgoing expiry height for htlc risk